### PR TITLE
helm: add extended-ceph-exporter as a helm dependency

### PR DESCRIPTION
**Pull request by ideepika**
Signed-off-by: Deepika Upadhyay <deepika@koor.tech>

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

**Checklist:**

- [ ] **Commit Message Formatting**: Commit titles and messages follow guidelines in the [developer guide](https://docs.koor.tech/docs/latest/Contributing/development-flow/#commit-structure)).
- [ ] **Skip Tests for Docs**: If this is only a documentation change, add the label `skip-ci` on the PR.
- [ ] Reviewed the developer guide on [Submitting a Pull Request](https://docs.koor.tech/docs/latest/Contributing/development-flow/#submitting-a-pull-request)
- [ ] [Pending release notes](https://github.com/koor-tech/koor/blob/master/PendingReleaseNotes.md) updated with breaking and/or notable changes for the next minor release.
- [ ] Documentation has been updated, if necessary.
- [ ] Unit tests have been added, if necessary.
- [ ] Integration tests have been added, if necessary.

### CODE REVIEW

**Review by galexrt — CHANGES_REQUESTED** [review]

File: deploy/charts/rook-ceph-cluster/values.yaml [hunk 1]
```diff
@@ -55,6 +55,17 @@ monitoring:
 # If true, create & use PSP resources. Set this to the same value as the rook-ceph chart.
 pspEnable: true
 
+extended-ceph-exporter:
```
Comment (galexrt) [inline deploy/charts/rook-ceph-cluster/values.yaml:58]:
```suggestion
extendedCephExporter:
```

Let's also add a short concise comment here and link to the extended-ceph-exporter Chart README.md

**Review by galexrt — CHANGES_REQUESTED** [review]

File: deploy/charts/rook-ceph-cluster/values.yaml (hunk 1 above)
Comment (ideepika) [reply deploy/charts/rook-ceph-cluster/values.yaml:58]: yeah that would be good! good catch, thanks Alex!

File: deploy/charts/rook-ceph-cluster/values.yaml [hunk 2]
```diff
@@ -55,6 +55,22 @@ monitoring:
...
+# `extended-ceph-exporter` rgw admin user with following config values, please
+# either set or update these values before helm dependency update.
+# This user will then be used to provide prometheus based extended ceph metrics.
+# See more: https://github.com/koor-tech/extended-ceph-exporter/blob/main/charts/extended-ceph-exporter/README.md
+extendedCephExporter:
+  enabled: true
+  config:
+  rgw:
+    # pass during helm install or set prior to it update
+    host: "https://your-ceph-rgw-endpoint-here"
```
Comment (galexrt) [inline deploy/charts/rook-ceph-cluster/values.yaml:68]: Could you please create an issue that we use Helm templating to auto generate this host address in the future when object store is enabled?

**Review by galexrt — APPROVED** [review]

File: build/makelib/helm.mk [hunk 3]
```diff
@@ -39,10 +39,14 @@ define helm.chart
 $(HELM_OUTPUT_DIR)/$(1)-$(VERSION).tgz: $(HELM) $(HELM_OUTPUT_DIR) $(shell find $(HELM_CHARTS_DIR)/$(1) -type f)
 	@echo === helm package $(1)
 	@rm -rf $(OUTPUT_DIR)/$(1)
+	@$(HELM) dependency build $(abspath $(HELM_CHARTS_DIR)/$(1))
 	@cp -aL $(HELM_CHARTS_DIR)/$(1) $(OUTPUT_DIR)
 	@$(SED_IN_PLACE) 's|VERSION|$(VERSION)|g' $(OUTPUT_DIR)/$(1)/values.yaml
 	@$(HELM) lint $(abspath $(OUTPUT_DIR)/$(1)) --set image.tag=$(VERSION)
-	$(HELM) package --version $(VERSION) --app-version $(VERSION) -d $(HELM_OUTPUT_DIR) $(abspath $(OUTPUT_DIR)/$(1))
+	@$(HELM) package --version $(VERSION) --app-version $(VERSION) -d $(HELM_OUTPUT_DIR) $(abspath $(OUTPUT_DIR)/$(1))
+	@rm -rf $(abspath $(HELM_CHARTS_DIR)/$(1))/Chart.lock \
```
Comment (galexrt) [inline build/makelib/helm.mk:47]: Why do we need to remove these files?

File: deploy/charts/rook-ceph-cluster/values.yaml [hunk 4]
```diff
@@ -55,6 +55,22 @@ monitoring:
...
 
+# If enabled, will install extended-ceph-exporter dependency and create
+# `extended-ceph-exporter` rgw admin user with following config values, please
+# either set or update these values before helm dependency update.
+# This user will then be used to provide prometheus based extended ceph metrics.
+# See more: https://github.com/koor-tech/extended-ceph-exporter/blob/main/charts/extended-ceph-exporter/README.md
+extendedCephExporter:
+  enabled: false
+  config:
+  rgw:
```
Comment (galexrt) [inline deploy/charts/rook-ceph-cluster/values.yaml:66]: Seems like the indentation is broken here

**Review by galexrt — APPROVED** [review]

**Pull request merged**